agent/eventhandler/handler.go

```diff
@@ -30,27 +30,20 @@ func HandleEngineEvents(taskEngine engine.TaskEngine, client api.ECSClient, save
 	statesaver = saver
 
 	for {
-		taskEvents, containerEvents := taskEngine.TaskEvents()
+		stateChangeEvents := taskEngine.StateChangeEvents()
```

Comment:
hm this pattern was in the original code. i think it was ensure we have a valid reference to channel before cycling through the inner loop to receive from the channel.